internal/controller/controllers/agent_controller.go

```diff
@@ -17,29 +17,193 @@ limitations under the License.
 package controllers
 
 import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/jinzhu/gorm"
+	"github.com/openshift/assisted-service/internal/bminventory"
+	"github.com/openshift/assisted-service/internal/common"
 	adiiov1alpha1 "github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/restapi/operations/installer"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // AgentReconciler reconciles a Agent object
 type AgentReconciler struct {
 	client.Client
-	Log    logrus.FieldLogger
-	Scheme *runtime.Scheme
+	Log       logrus.FieldLogger
+	Scheme    *runtime.Scheme
+	Installer bminventory.InstallerInternals
 }
 
 // +kubebuilder:rbac:groups=adi.io.my.domain,resources=agents,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=adi.io.my.domain,resources=agents/status,verbs=get;update;patch
 
 func (r *AgentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	// your logic here
+	ctx := context.Background()
+	agent := &adiiov1alpha1.Agent{}
+	var Requeue bool
+	var inventoryErr error
+
+	err := r.Get(ctx, req.NamespacedName, agent)
+	if err != nil {
+		r.Log.WithError(err).Errorf("Failed to get resource %s", req.NamespacedName)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	if agent.Spec.ClusterDeploymentName == nil {
```

Comment:
Maybe host should be just ignored?
It will be valid once we have late binding

Reply:
Ack. Ignored the Agent in this case.